### PR TITLE
Create tests for ListSets [ch140]

### PR DESCRIPTION
**Pull request by ConnorSheremeta**
#1596

Details here https://github.com/ualbertalib/jupiter/pull/1427

### CODE REVIEW

File: test/fixtures/items.yml [hunk 1]
```diff
@@ -19,6 +20,20 @@ admin:
...
   created: 'Winter 2017'
   record_created_at: <%= 5.day.ago.to_s(:db) %>
+  updated_at: <%= 5.day.from_now.to_s(:db) %>
+  languages: <%= [CONTROLLED_VOCABULARIES[:language].english] %>
+  license: <%= CONTROLLED_VOCABULARIES[:license].attribution_4_0_international %>
+  item_type: <%= CONTROLLED_VOCABULARIES[:item_type].book %>
+  subject: ['Ownership']
+  member_of_paths: <%= ["#{ActiveRecord::FixtureSet.identify(:fancy_community, :uuid)}/#{ActiveRecord::FixtureSet.identify(:fancy_collection, :uuid)}"] %>
+
+private_item:
```
Comment (murny) [inline test/fixtures/items.yml:30]: Looks like a bad merge or something? These diffs already exist. In fact it looks like Private Item is now duplicated twice in yours:
https://github.com/ualbertalib/jupiter/blob/cf5e167efe8c26670bf2b065f56c013068619322/test/fixtures/items.yml#L30-L55

File: test/fixtures/thesis.yml [hunk 2]
```diff
@@ -15,5 +27,28 @@ private:
   dissertant: 'Joe Blow'
   graduation_date: '2019'
   record_created_at: <%= 5.day.ago.to_s(:db) %>
+  updated_at: <%= 10.day.ago.to_s(:db) %>
   abstract: 'This is my abstract'
   member_of_paths: <%= ["#{ActiveRecord::FixtureSet.identify(:thesis, :uuid)}/#{ActiveRecord::FixtureSet.identify(:thesis, :uuid)}"] %>
+
+fancy:
```
Comment (murny) [inline test/fixtures/thesis.yml:34]: Why we bringing in all these new fixtures? Don't see these being used in your tests?

File: test/fixtures/collections.yml [hunk 3]
```diff
@@ -13,9 +13,24 @@ thesis:
...
+  date_ingested: <%= Time.zone.now - 1.hour %>
+
 fancy_collection:
   title: 'Fancy Collection'
   owner: admin
+  community: thesis
+  record_created_at: <%= Time.zone.now - 1.hour %>
+  date_ingested: <%= Time.zone.now - 1.hour %>
+
+fancier_collection:
```
Comment (murny) [inline test/fixtures/collections.yml:31]: Same thing here, This is now duplicated twice: https://github.com/ualbertalib/jupiter/blob/cf5e167efe8c26670bf2b065f56c013068619322/test/fixtures/collections.yml#L31-L43


Something strange going on 🤔

File: test/fixtures/thesis.yml (hunk 2 above)
Comment (ConnorSheremeta) [reply test/fixtures/thesis.yml:34]: They're used in oaisys_list_identifiers_test.rb to test from, until, and from/until parameters which is based off of updated_at

Comment (ConnorSheremeta) [reply test/fixtures/thesis.yml:34]: Needed three theses to do a proper test (private and embargoed won't appear in the response)

Comment (murny) [reply test/fixtures/thesis.yml:34]: Sounds good was just checking 👍

File: test/fixtures/collections.yml [hunk 4]
```diff
@@ -13,10 +13,18 @@ thesis:
...
+  restricted: true
+  owner: admin
+  community: thesis
+  record_created_at: <%= Time.zone.now - 1.hour %>
+  date_ingested: <%= Time.zone.now - 1.hour %>
+
 fancy_collection:
   title: 'Fancy Collection'
   owner: admin
-  community: fancy_community
```
Comment (murny) [inline test/fixtures/collections.yml:19]: maybe should keep this as it was? Some test might depend on it which might account for the failing tests on CI 🤔

Comment (ConnorSheremeta) [reply test/fixtures/collections.yml]: Thanks for catching this, this accounted for the failing tests. I still have one failing locally regardless of what branch I'm on? I'll just ignore that for now.

Comment (murny) [reply test/fixtures/collections.yml]: Oh weird? Which test is it? Maybe just flaking 🤔

Comment (ConnorSheremeta) [reply test/fixtures/collections.yml]: F

Failure:
Items::FilesControllerTest#test_should_be_able_to_attach_files_to_a_draft_item [/home/connor/jupiter_master/test/controllers/items/files_controller_test.rb:27]:
"ActiveStorage::Attachment.count" didn't change by 1.
Expected: 2
  Actual: 3

**Comment by ConnorSheremeta** [comment]
> Got no context around the OASIS stuff, but Code looks good to me

Do you think Matt should have a quick look over this or is this good to merge?

File: test/fixtures/collections.yml (hunk 4 above)
Comment (murny) [reply test/fixtures/collections.yml]: Wonder if anyone else has this problem? My tests pass just like CI so can't reproduce on my end 🤔. I guess if you really want to know you can put a debug statement and see why its not adding an attachment to the object (maybe validation or something weird going on)

**Pull request merged**